src/SimpleGraphs/simpledigraph.jl

```diff
@@ -82,6 +82,146 @@ function SimpleDiGraph(g::AbstractSimpleGraph)
     return h
 end
 
+
+@inbounds function cleanupedges!(fadjlist::Vector{Vector{T}},
+                                 badjlist::Vector{Vector{T}}) where T<:Integer
+    neg = 0
+    for v in 1:length(fadjlist)
+        if !issorted(fadjlist[v])
+            sort!(fadjlist[v])
```

Comment:
this check can be expensive. Might as well just sort and be done with it, or pass in a boolean argument to denote whether we can assume the vectors are sorted.

Reply:
I don't think calling `issorted` is that expensive; especially compared to `sort! `. When I wrote that function I used the profiler to check where the bottlenecks are and `issort`  was definitely not the problem.  `unique!` actually also calls this function, which is unnecessary in our case as we know that the vector must be sorted. 

I could implement a flag to indicate that the input  is sorted and otherwise correct but then the problem is that a user could actually produce an invalid graph. I think, if someone really need to optimize for the last bits of performance, then they are better of by creating fadjlist and badjlist themself. I tested that by writing a function that builds a complete graph and for 10^4 vertices I am a factor of 2.5 times faster than by creating a vector of edges and calling SimpleGraph with sorting disabled. 

Then I wrote some code to see what the performance is, if we don't sort/don't check for sorted input:
Here are the results for 100 and 10^4 vertices:
```
julia> run_speed_tests(100; ntimes=50)
Sorted Input:
  SimpleGraph                  : mean == 0.000229s  std == 1.61e-5s
  SimpleGraph (no checks)       : mean == 0.00258s  std == 0.0174s
  SimpleGraph (always sort)    : mean == 0.000119s  std == 2.1e-5s
Shuffled Input:
  SimpleGraph                  : mean == 0.000307s  std == 2.2e-5s
  SimpleGraph (always sort)    : mean == 0.000325s  std == 2.03e-5s
```
```
julia> run_speed_tests(10000; ntimes=50)
Sorted Input:
  SimpleGraph                  : mean == 1.44s  std == 0.0452s
  SimpleGraph (no checks)       : mean == 1.16s  std == 0.0314s
  SimpleGraph (always sort)    : mean == 1.97s  std == 0.0348s
Shuffled Input:
  SimpleGraph                  : mean == 7.16s  std == 0.079s
  SimpleGraph (always sort)    : mean == 7.26s  std == 0.068s
```
- Sorted input constructs a complete graph from a sorted list of edges
- Shuffled Input constructs the same graph when the list is randomly shuffled

- Simple graph is the constructor form the pr
- Simple graph (no checks) assumes the input is sorted, has no duplicates and the vertices of each edge have values >= 1.
- Simple graph (always sort) does not check if the vertices are sorted and just sorts them

As you can see, if the input is random, then `issort` does not make the code much slower, at least not for large graphs. If the input is already sorted, then using  `issort` is faster than just sorting. In this case, using a flag that indicates sorted input gives some performance improvements  but as I said, I'm not sure if it is worth the potential errors that improper usage could create.

Reply:
The numbers don't lie. Thank you for checking. `issorted` will be at worst an O(n) operation, right?

I agree that the performance difference is not significant enough to worry about at this point. Carry on!

Reply:
Yes, O(n) where n is the number of input edges.